torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -369,6 +379,9 @@ std::shared_ptr<FutureMessage> RequestCallbackImpl::processRpc(
 
 std::shared_ptr<FutureMessage> RequestCallbackImpl::processMessage(
     Message& request) const {
+  if (request.hasUserFunction()) {
+    RRefContext::getInstance().recordThreadLocalPendingUsers();
```

Comment:
Shouldn't this apply to all Messages? If in the future I build a custom message for internal system messages and send an RRef in it, we should still honor the same protocol where we wait for confirmation right?

Reply:
Sure, will make the change.